File: 백준/Gold/12904. A와 B/A와 B.py
S = list(input())
T = list(input())

while len(S) < len(T):
    if T[-1] == 'A':
        T.pop(-1)
    else:
        T.pop(-1)
        T.reverse()

if S == T:
    print(1)
else:
    print(0)

File: 백준/Gold/12904. A와 B/A와 B.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws Exception {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		StringBuilder S = new StringBuilder(br.readLine());
		StringBuilder T = new StringBuilder(br.readLine());
		
		while(T.length() > S.length()) {
			if(T.charAt(T.length()-1) == 'A') {
				T.deleteCharAt(T.length()-1);
			} else {
				T.deleteCharAt(T.length()-1);
				T.reverse();
			}
		}
		
		if(T.toString().equals(S.toString())) System.out.println(1);
		else System.out.println(0);
	}

}
